kmod/src/net.h

```diff
@@ -90,19 +90,20 @@ enum conn_flags {
 #define SIN_ARG(sin)	sin, be16_to_cpu((sin)->sin_port)
 
 static inline void scoutfs_addr_to_sin(struct sockaddr_in *sin,
-				       struct scoutfs_inet_addr *addr)
+				       union scoutfs_inet_addr *addr)
 {
 	sin->sin_family = AF_INET;
-	sin->sin_addr.s_addr = cpu_to_be32(le32_to_cpu(addr->addr));
-	sin->sin_port = cpu_to_be16(le16_to_cpu(addr->port));
+	sin->sin_addr.s_addr = cpu_to_be32(le32_to_cpu(addr->v4.addr));
+	sin->sin_port = cpu_to_be16(le16_to_cpu(addr->v4.port));
 }
 
-static inline void scoutfs_addr_from_sin(struct scoutfs_inet_addr *addr,
+static inline void scoutfs_addr_from_sin(union scoutfs_inet_addr *addr,
 					 struct sockaddr_in *sin)
 {
-	addr->addr = be32_to_le32(sin->sin_addr.s_addr);
-	addr->port = be16_to_le16(sin->sin_port);
-	memset(addr->__pad, 0, sizeof(addr->__pad));
+	memset(addr, 0, sizeof(union scoutfs_inet_addr));
+	addr->v4.family = sin->sin_family;
```

Comment:
We can't rely on the family constants being consistent across ABIs.  We need our own family enum and a table to go between the two, sort of like we do for transporting errnos as message error responses.  But hilariously enough, this function isn't used anymore!  It was only used to store the server's bound address in the super for clients to connect to.  Now the clients connect to known addresses written by mkfs.  The kernel never writes addresses into the quorum config.  So we can just remove this function.